lsql/judge/tests/test_submit.py

```diff
@@ -241,3 +243,60 @@ def test_discriminant_problem(self):
                                follow=True)
         self.assertEqual(response.json()['veredict'], VeredictCode.RE)
         self.assertEqual(disc_problem.problem_type(), ProblemType.DISC)
+
+    def test_achievements_submit(self):
+        """Test to show correct message when obtain an achievement"""
+        client = Client()
+        collection = create_collection('Colleccion de prueba XYZ')
+        select_problem = create_select_problem(collection, 'SelectProblem ABC DEF')
+        user = create_user('5555', 'tamara')
+        ach_submission = NumSubmissionsProblemsAchievementDefinition(name={"es": 'Un envio'},
+                                                                      description={
+                                                                          "es": 'Envia una solucion para un problema'},
+                                                                      num_problems=1, num_submissions=1)
+        ach_submission.save()
+        ach_submissions = NumSubmissionsProblemsAchievementDefinition(name={"es": 'Tres envios'},
+                                                                      description={
+                                                                          "es": 'Envia tres soluciones de un problema'},
+                                                                      num_problems=1, num_submissions=3)
+        ach_submissions.save()
+        ach_type = NumSolvedTypeAchievementDefinition(name={"es": 'Es select'},
+                                                      description={"es": 'Resuelve un problema SELECT'},
+                                                      num_problems=1, problem_type=ProblemType.SELECT.name)
+        client.login(username='tamara', password='5555')
+        submit_select_url = reverse('judge:submit', args=[select_problem.pk])
+
+        # The user submits one solution and obtains the first achievement
+        response = client.post(submit_select_url, {'code': 'MAL'}, follow=True)  # Validation Error, too short
+        obtained_achieve = ObtainedAchievement.objects.filter(user=user)
+        self.assertIn( obtained_achieve[0].achievement_definition.name['es'], response.json()['achievements'])
+
+        # The user submits a new solution and does not receive any achievement
+        response = client.post(submit_select_url, {'code': 'MAL'}, follow=True)  # Validation Error, too short
+        try:
+            response.json()['achievements']
+        except KeyError:
+            msg = 'no exixte'
+            self.assertEqual('no exixte', msg)
```

Comment:
El aserto "no exixte" = "no exixte" es poco útil. Lo que querrás comprobar es que `'achievements' not in response.json()` con un aserto `assertNotIn`.